client/signup/main.jsx

```diff
@@ -185,13 +191,13 @@ const Signup = React.createClass( {
 	componentDidMount() {
 		debug( 'Signup component mounted' );
 		SignupProgressStore.on( 'change', this.loadProgressFromStore );
-		SignupProgressStore.on( 'change', this.loadDependenciesFromStore );
+		this.unsubscribeSignupDepStoreListener = SignupDependencyStore.reduxStore.subscribe( this.loadDependenciesFromStore );
```

Comment:
It is possible to check if there have been any real changes by comparing `this.state.dependencies` with what is in the Redux state. This would save extra re-renders, but will take a bit more of the CPU since it has to do a deep comparison of both objects (e.g. `_.isEqual(this.state.dependencies, signupDependencyStoreFromRedux)`). 

I am not sure which would be better. Any ideas?